### PR TITLE
Remove promise kit

### PR DESCRIPTION
**Pull request by joewalsh**

### CODE REVIEW

File: Wikipedia/Code/SavedArticlesFetcher.m [hunk 1]
```diff
@@ -342,6 +342,7 @@ - (void)fetchImageInfoForImagesInArticle:(MWKArticle *)article failure:(WMFError
             }
             success:^(id _Nonnull object) {
                 if (!object || [object isEqual:[NSNull null]]) {
+                    [group leave];
```
Comment (coreyfloyd) [inline Wikipedia/Code/SavedArticlesFetcher.m]: Oh man - Has this been like this?

File: Wikipedia/Code/WMFArticleFetcher.m [hunk 2]
```diff
@@ -88,6 +88,7 @@ - (nullable NSURLSessionTask *)fetchArticleForURL:(NSURL *)articleURL
                                           success:(WMFArticleHandler)success {
     if (!articleURL.wmf_title) {
         failure([NSError wmf_errorWithType:WMFErrorTypeStringMissingParameter userInfo:nil]);
+        return nil;
```
Comment (coreyfloyd) [inline Wikipedia/Code/WMFArticleFetcher.m]: Was this also a pre-existing bug?

**Comment by coreyfloyd** [comment]
@joewalsh looks good. Were those missing returns/saves all previous issues or just since the conversion?

File: Wikipedia/Code/WMFMainPageContentSource.m [hunk 3]
```diff
@@ -92,6 +92,7 @@ - (void)fetchAndSaveMainPageForSection:(WMFContentGroup *)section completion:(nu
         if (completion) {
             completion();
         }
+        return;
```
Comment (joewalsh) [inline Wikipedia/Code/WMFMainPageContentSource.m]: @coreyfloyd this was an existing bug

File: Wikipedia/Code/WMFArticleFetcher.m (hunk 2 above)
Comment (joewalsh) [inline Wikipedia/Code/WMFArticleFetcher.m]: @coreyfloyd  this was a bug in conversion

File: Wikipedia/Code/SavedArticlesFetcher.m (hunk 1 above)
Comment (joewalsh) [inline Wikipedia/Code/SavedArticlesFetcher.m]: @coreyfloyd this was a bug in conversion

**Comment by montehurd** [comment]
For the record, this is awesome :)

**Pull request merged**